pkg/sidecar/config/config.go

```diff
@@ -243,8 +248,73 @@ func (s *ScyllaConfig) setupEntrypoint(ctx context.Context) (*exec.Cmd, error) {
+
+		// Determine expected IP family based on broadcast-address
+		isBroadcastIPv6 := strings.Contains(m.BroadcastAddress, ":")
+
+		// Validate rpc-address IP family consistency if user provided it
+		if userRpcAddress, hasUserRpcAddress := userArgs["rpc-address"]; hasUserRpcAddress && userRpcAddress != nil {
+			isUserRpcIPv6 := strings.Contains(*userRpcAddress, ":")
+
+			if isBroadcastIPv6 != isUserRpcIPv6 {
+				// IP families don't match, reject user's value and log warning
+				klog.Warningf("User-provided rpc-address '%s' IP family doesn't match broadcast-address '%s' IP family. Removing user's rpc-address for consistency.",
+					*userRpcAddress, m.BroadcastAddress)
+				// Remove user's rpc-address so we'll use operator default
+				delete(userArgs, "rpc-address")
+			} else {
+				klog.Infof("Using user-provided rpc-address: %s", *userRpcAddress)
+			}
+		}
+
+		// Validate listen-address IP family consistency if user provided it
+		if userListenAddress, hasUserListenAddress := userArgs["listen-address"]; hasUserListenAddress && userListenAddress != nil {
+			isUserListenIPv6 := strings.Contains(*userListenAddress, ":")
+
+			if isBroadcastIPv6 != isUserListenIPv6 {
+				// IP families don't match, reject user's value and log warning
+				klog.Warningf("User-provided listen-address '%s' IP family doesn't match broadcast-address '%s' IP family. Removing user's listen-address for consistency.",
+					*userListenAddress, m.BroadcastAddress)
+				// Remove user's listen-address so we'll use operator default
+				delete(userArgs, "listen-address")
+			} else {
+				klog.Infof("Using user-provided listen-address: %s", *userListenAddress)
+			}
+		}
+
+		// Merge user arguments with operator-managed arguments
+		args = mergeArguments(args, userArgs)
+	}
+
+	// Set default addresses only if not provided by user (or were rejected due to IP family mismatch)
+	// Make sure that all addresses use the same IP family as broadcast-address for consistency
+	isBroadcastIPv6 := strings.Contains(m.BroadcastAddress, ":")
+
+	if _, hasRpcAddress := args["rpc-address"]; !hasRpcAddress {
+		defaultRpcAddress := "0.0.0.0"
+		if isBroadcastIPv6 {
+			defaultRpcAddress = "::"
+		}
+		args["rpc-address"] = &defaultRpcAddress
+		klog.Infof("Using default rpc-address: %s (IP family matches broadcast-address)", defaultRpcAddress)
+	}
+
+	if _, hasListenAddress := args["listen-address"]; !hasListenAddress {
+		defaultListenAddress := "0.0.0.0"
+		if isBroadcastIPv6 {
+			defaultListenAddress = "::"
+		}
+		args["listen-address"] = &defaultListenAddress
+		klog.Infof("Using default listen-address: %s (IP family matches broadcast-address)", defaultListenAddress)
+	}
+
+	// Enable IPv6 DNS lookups if using IPv6 broadcast address to resolve IPv6 DNS resolution issues
+	if isBroadcastIPv6 {
+		args["enable-ipv6-dns-lookup"] = pointer.Ptr("1")
+		klog.Infof("Enabling IPv6 DNS lookup due to IPv6 broadcast address: %s", m.BroadcastAddress)
```

Comment:
The broadcast address is a result of configuring the cluster with `IPFamily=IPv6`. I'm not sure if we should communicate that the IPv6 DNS lookup is caused by the broadcast address being IPv6, as it's only an indirect indicator of the IPv6 setting for the whole cluster.

Maybe something like this instead?

```suggestion
		klog.Info("Enabling IPv6 DNS lookup due to cluster IPv6 IPFamily")
```